### PR TITLE
Equation DB

### PR DESCRIPTION
**Pull request by paul841029**
screenshot from 2019-02-15 09-45-52

Already done:
1. Added code for extracting equation into a separate table. The original copy in the sentence table is not removed.
2. Added dependencies of im2latex and latex parser to requirement.txt at the top level. https://github.com/UW-COSMOS/Cosmos/blob/latex/exp/requirements.txt
3. Embedded the API call for im2latex in list2html. https://github.com/UW-COSMOS/Cosmos/blob/latex/exp/list2html.py#L49
4. Embedded the API call for the variable parser in the equation table routine.  https://github.com/UW-COSMOS/Cosmos/blob/latex/exp/Parser/insert_equation.py#L78
5. Tested end-to-end from docker.
   
Reference for procedures. https://docs.google.com/document/d/18YVWqQVfG0MJN8KjFXVYwGCY0pbFZSxm7W9wGBj8KeE/edit

### CODE REVIEW

**Comment by paul841029** [comment]
Ready for review and merge. @ankur-gos

File: cosmos/Parser/insert_equation.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+INT_ARRAY_TYPE = postgresql.ARRAY(Integer)
+STR_ARRAY_TYPE = postgresql.ARRAY(String)
+
+# Grab pointer to global metadata
+db_connect_str = "postgres://postgres:password@localhost:5432/cosmos8"
+_meta = Meta.init()
+
+
+class Equation(_meta.Base):
```
Comment (ankur-gos) [inline cosmos/Parser/insert_equation.py]: Would prefer for this class declaration to be in its ownf ile

File: cosmos/Parser/insert_equation.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+                        'page': sent.page
+                    }
+                )
+
+        for sec_id, para_dic in eqs_groupby_section_para.items():
+            for paragraph_id, eqs in para_dic.items():
+                latex_code = ''.join(map(lambda x: x['text'], eqs))
+                variables = list(get_variables(latex_code))
+                if len(variables) == 0 or variables[0] == -1:
+                    variables = None
```
Comment (ankur-gos) [inline cosmos/Parser/insert_equation.py]: Should we insert equations that have no variables? Should every equation have at least 1 variable?

File: cosmos/Parser/latex_parser/variable_extractor.py [hunk 3]
```diff
@@ -28,14 +28,12 @@ def get_variables(latex):
     processed = processed.replace(')', '}')
 
     response = muterun_js(os.path.join(CURRENT_DIR, 'katex/parse.js'), processed)
-    # print(response.stderr)
     if response.stdout == b'-1\n' or not response.stdout:
         yield -1
         return
 
     tree = etree.fromstring(response.stdout)
     ast = tree.xpath('.//semantics')[0]
```
Comment (ankur-gos) [inline cosmos/Parser/latex_parser/variable_extractor.py]: Would prefer this area to be commented a little more and have descriptive variables. I'm not quite sure what's going on here.

**Pull request merged**